.github/workflows/ci.yml

```diff
@@ -234,35 +234,58 @@ jobs:
     runs-on: hosted-windows-1
     steps:
       # more info here:- https://github.com/rust-lang/cargo/issues/13020
+      - name: Create Dev Drive using ReFS
+        run: |
+          $Volume = New-VHD -Path C:/zed_dev_drive.vhdx -SizeBytes 30GB |
+                    Mount-VHD -Passthru |
+                    Initialize-Disk -Passthru |
+                    New-Partition -AssignDriveLetter -UseMaximumSize |
+                    Format-Volume -FileSystem ReFS -Confirm:$false -Force
+          Write-Output $Volume
+          Write-Output "DEV_DRIVE=$($Volume.DriveLetter):" >> $env:GITHUB_ENV
       - name: Enable longer pathnames for git
         run: git config --system core.longpaths true
       - name: Checkout repo
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
         with:
           clean: false
+      # actions/checkout does not let us clone into anywhere outside ${{ github.workspace }}, so we have to copy the clone...
+      - name: Copy Git Repo to Dev Drive
+        run: |
+          Copy-Item -Path "${{ github.workspace }}" -Destination "${{ env.DEV_DRIVE }}/zed" -Recurse
 
       - name: Cache dependencies
         uses: swatinem/rust-cache@f0deed1e0edfc6a9be95417288c0e1099b1eeec3 # v2
         with:
-          save-if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
We should bring this check back. I got rid of it only to see if it helps anyhow with this PR.